src/HealthCheck/HttpHealthCheck.php

```diff
@@ -0,0 +1,160 @@
+<?php
+declare(strict_types=1);
+
+namespace ShoppinPal\YapepCommon\HealthCheck;
+
+use ShoppinPal\YapepCommon\Exception\HealthCheckException;
+use YapepBase\Application;
+use YapepBase\Communication\CurlHttpRequest;
+use YapepBase\Communication\CurlHttpRequestResult;
+use YapepBase\Config;
+use YapepBase\Exception\ConfigException;
+use YapepBase\Exception\CurlException;
+use YapepBase\Exception\ParameterException;
+
+class HttpHealthCheck implements IHealthCheck
+{
+    /**
+     * @throws ParameterException
+     * @throws HealthCheckException
+     * @throws ConfigException
+     * @throws CurlException
+     */
+    public function checkServiceHealth(array $configOptions): void
+    {
+        $url          = $this->getUrl($configOptions);
+        $result       = $this->sendRequest($configOptions, $url);
+        $responseBody = $result->getResponseBody();
+
+        $this->checkStatus($configOptions, $result, $url);
+        $this->checkJson($configOptions, $responseBody);
+        $this->checkBody($configOptions, $responseBody);
+    }
+
+    protected function checkArrayContents(array $expected, array $array): bool
```

Comment:
This method is not clear to me (can you find a better name?), but it looks somethign general what should go into another class.
Also some unit tests would be awesome for this

Reply:
Some tests in general would be good, so we should probably test the whole project properly.